kornia/feature/keynet.py

```diff
@@ -0,0 +1,315 @@
+    """
+
+    def __init__(self, in_channels: int = 10):
+        super().__init__()
+
+        self.conv0 = conv_blck(in_channels)
+        self.conv1 = conv_blck()
+        self.conv2 = conv_blck()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.conv2(self.conv1(self.conv0(x)))
+        return x
+
+
+def conv_blck(in_channels: int = 8,
+              out_channels: int = 8,
+              kernel_size: int = 5,
+              stride: int = 1,
+              padding: int = 2,
+              dilation: int = 1):
+    """Helper function for KeyNet.
+
+    Default learnable convolutional block for KeyNet.
+    """
+    return nn.Sequential(nn.Conv2d(in_channels, out_channels, kernel_size,
+                                   stride, padding, dilation),
+                         nn.BatchNorm2d(out_channels),
+                         nn.ReLU(inplace=True))
+
+
+class KeyNet(nn.Module):
+    """Key.Net model definition -- local feature detector (response function). This is based on the original code
+    from paper "Key.Net: Keypoint Detection by Handcrafted and Learned CNN Filters". See :cite:`KeyNet2019` for
+    more details.
+
+    Args:
+        pretrained: Download and set pretrained weights to the model.
+        keynet_conf: Dict with initiliazation parameters. Do not pass it, unless you know what you are doing`.
+
+    Returns:
+        torch.Tensor: KeyNet response score.
+
+    Shape:
+        - Input: :math:`(B, 1, H, W)`
+        - Output: :math:`(B, 1, H, W)`
+    """
+    def __init__(self,
+                 pretrained: bool = False,
+                 keynet_conf: Dict = keynet_config['KeyNet_default_config']):
+        super().__init__()
+
+        num_filters = keynet_conf['num_filters']
+        self.num_levels: int = keynet_conf['num_levels']
+        kernel_size: int = keynet_conf['kernel_size']
+        padding: int = kernel_size // 2
+
+        self.feature_extractor = feature_extractor()
+        self.last_conv = nn.Sequential(nn.Conv2d(in_channels=num_filters * self.num_levels,
+                                                 out_channels=1, kernel_size=kernel_size, padding=padding),
+                                       nn.ReLU(inplace=True))
+        # use torch.hub to load pretrained model
+        if pretrained:
+            pretrained_dict = torch.hub.load_state_dict_from_url(
+                urls['keynet'], map_location=lambda storage, loc: storage
+            )
+            self.load_state_dict(pretrained_dict['state_dict'], strict=True)
+        self.eval()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        x - input image
+        """
+        shape_im = x.shape
+        feats: List[torch.Tensor] = [self.feature_extractor(x)]
+        for i in range(1, self.num_levels):
+            x = pyrdown(x, factor=1.2)
+            feats_i = self.feature_extractor(x)
+            feats_i = F.interpolate(feats_i, size=(shape_im[2], shape_im[3]), mode='bilinear')
```

Comment:
reuse `kornia.geometry.resize` ?

Reply:
Given that KeyNet was not trained with 'kornia.geometry.resize', it might affect the quality